### PR TITLE
[PB-2122, PB-2119] Feat: Get Payment Methods - Cancel B2B Subscriptions

### PR DESCRIPTION
**Pull request by evillalba94**
#### Update Webhook Events
Changes were made to handle B2B subscriptions events, including:
    - deleting the workspace in case of canceling the subscription
PR related: https://github.com/internxt/drive-server-wip/pull/329

### CODE REVIEW

**Comment by evillalba94** [comment]
@sg-gs 
Could you help me with the review?

File: src/services/PaymentService.ts [hunk 1]
```diff
@@ -341,8 +341,21 @@ export class PaymentService {
    *  customer.invoice_settings.default_payment_method that precedence over
    *  customer.default_source
    */
-  async getDefaultPaymentMethod(customerId: string): Promise<PaymentMethod | CustomerSource | null> {
-    const subscriptions = await this.getActiveSubscriptions(customerId);
+  async getDefaultPaymentMethod(
+    customerId: CustomerId,
+    subscriptionType: 'individual' | 'business' = 'individual',
+  ): Promise<PaymentMethod | CustomerSource | null> {
+    if (!customerId)
```
Comment (sg-gs) [inline src/services/PaymentService.ts:348]: Remove this, as the customerId type indicates that it is not null. Any additional validation should be done in the controller.

File: src/services/PaymentService.ts [hunk 2]
```diff
@@ -341,8 +341,21 @@ export class PaymentService {
...
+      throw new Error('customerId required');
+
+    let subscriptions = await this.getActiveSubscriptions(customerId);
+    if (!subscriptions.length)
```
Comment (sg-gs) [inline src/services/PaymentService.ts:352]: Use `subscriptions.length === 0`

File: src/services/UsersService.ts [hunk 3]
```diff
@@ -82,21 +82,39 @@ export class UsersService {
...
+    if (!customerId) throw new Error('customerId required');
     const activeSubscriptions = await this.paymentService.getActiveSubscriptions(customerId);
 
     if (activeSubscriptions.length === 0) {
       throw new Error('Subscriptions not found');
     }
 
     const individualSubscriptions = activeSubscriptions.filter(
-      (subscription) => subscription.metadata.is_teams !== '1',
+      (subscription) => subscription.metadata.is_teams !== '1' && subscription.product?.metadata.type !== 'business',
```
Comment (sg-gs) [inline src/services/UsersService.ts:93]: Remove the `is_teams` flag usage, we only use our current flags, not legacy ones

File: src/services/UsersService.ts [hunk 4]
```diff
@@ -82,21 +82,39 @@ export class UsersService {
...
     ) as Stripe.Subscription[];
 
     for (const subscriptionToCancel of individualSubscriptions) {
       await this.paymentService.cancelSubscription(subscriptionToCancel.id);
     }
   }
 
+  async cancelUserB2BSuscriptions(customerId: User['customerId']): Promise<void> {
+    if (!customerId) throw new Error('customerId required');
```
Comment (sg-gs) [inline src/services/UsersService.ts:102]: This is the service, the type does not indicate that the customerId is null or undefined, validate it on the controller

**Review by sg-gs — APPROVED** [review]
Approved, but review the comments @evillalba94

**Pull request merged**